### DISCUSSION TITLE
More Streamlined Search Filters

### DISCUSSION THREAD

**Discussion by EpiCreeper** [Quaver/Quaver#4767] [Feature Request]
### Is your feature request related to a problem? Please describe.

Currently, searching for maps with specific attributes either requires the use of text flags, only present to players through the Quaver wiki, or the "SORT BY" dropdown menu, that sorts ALL of your, maps without filtering any, while also allowing only one option selected at a time.
It's obvious that the text flags were implemented serve that niche, but keeping them exclusive to the quaver wiki makes it difficult for players to learn about and use them. Most new players don't even know about the quaver wiki or care to read it, and as a non native English speaker, I often forget the keywords and have to go search for them.

### Describe the solution you'd like

I would like a button next to the search bar, showing a menu similar to the table in the wiki. It would show all the filter options and keywords. It would also show all the usable operators, such as = or <. Pressing on any of them would add the filter to the search bar. I've made a mockup of it:
![11232023 15-59-58-741](https://github.com/Quaver/Quaver/assets/77532703/3e2a65a4-15dc-47b1-b228-b465771032c5)

### Describe alternatives you've considered

I think just adding an option to see the filters, without the added functionality, could also be very useful. Alternatively, the search bar itself could act as a dropdown when selected, similar to typing '/' in discord.

### Additional context

IDK what to do here...

**Comment by WilliamQiufeng** [comment]
imo a filter menu similar to the one in mapset download screen would also be nice. 
P.S. just thinking: Maybe we can also support grouping for artists / difficulties?
